tests/test_compute_funs.py

```diff
@@ -100,83 +100,147 @@ def test_magnetic_field_derivatives(DummyStellarator):
+    B_sup_zeta_rr = np.convolve(data["B^zeta"], FD_COEF_2_4, "same") / drho**2
     B_sub_rho_r = np.convolve(data["B_rho"], FD_COEF_1_4, "same") / drho
+    B_sub_rho_rr = np.convolve(data["B_rho"], FD_COEF_2_4, "same") / drho**2
     B_sub_theta_r = np.convolve(data["B_theta"], FD_COEF_1_4, "same") / drho
+    B_sub_theta_rr = np.convolve(data["B_theta"], FD_COEF_2_4, "same") / drho**2
     B_sub_zeta_r = np.convolve(data["B_zeta"], FD_COEF_1_4, "same") / drho
+    B_sub_zeta_rr = np.convolve(data["B_zeta"], FD_COEF_2_4, "same") / drho**2
+    B_r = np.convolve(data["|B|"], FD_COEF_1_4, "same") / drho
+    B_rr = np.convolve(data["|B|"], FD_COEF_2_4, "same") / drho**2
 
     np.testing.assert_allclose(
         data["B^theta_r"][3:-2],
         B_sup_theta_r[3:-2],
-        rtol=1e-3,
-        atol=1e-3 * np.nanmean(np.abs(data["B^theta_r"])),
+        rtol=rtol,
+        atol=atol * np.nanmean(np.abs(data["B^theta_r"])),
+    )
+    np.testing.assert_allclose(
+        data["B^theta_rr"][3:-2],
+        B_sup_theta_rr[3:-2],
+        rtol=rtol,
+        atol=atol * np.nanmean(np.abs(data["B^theta_rr"])),
     )
     np.testing.assert_allclose(
         data["B^zeta_r"][3:-2],
         B_sup_zeta_r[3:-2],
-        rtol=1e-3,
-        atol=1e-3 * np.nanmean(np.abs(data["B^zeta_r"])),
+        rtol=rtol,
+        atol=atol * np.nanmean(np.abs(data["B^zeta_r"])),
+    )
+    np.testing.assert_allclose(
+        data["B^zeta_rr"][3:-2],
+        B_sup_zeta_rr[3:-2],
+        rtol=rtol,
+        atol=atol * np.nanmean(np.abs(data["B^zeta_rr"])),
     )
     np.testing.assert_allclose(
         data["B_rho_r"][3:-2],
         B_sub_rho_r[3:-2],
-        rtol=1e-3,
-        atol=1e-3 * np.nanmean(np.abs(data["B_rho_r"])),
+        rtol=rtol,
+        atol=atol * np.nanmean(np.abs(data["B_rho_r"])),
+    )
+    np.testing.assert_allclose(
+        data["B_rho_rr"][3:-2],
+        B_sub_rho_rr[3:-2],
+        rtol=rtol,
+        atol=atol * np.nanmean(np.abs(data["B_rho_rr"])),
     )
     np.testing.assert_allclose(
         data["B_theta_r"][3:-2],
         B_sub_theta_r[3:-2],
-        rtol=1e-3,
-        atol=1e-3 * np.nanmean(np.abs(data["B_theta_r"])),
+        rtol=rtol,
+        atol=atol * np.nanmean(np.abs(data["B_theta_r"])),
+    )
+    np.testing.assert_allclose(
+        data["B_theta_rr"][3:-2],
+        B_sub_theta_rr[3:-2],
+        rtol=rtol,
+        atol=atol * np.nanmean(np.abs(data["B_theta_rr"])),
     )
     np.testing.assert_allclose(
         data["B_zeta_r"][3:-2],
         B_sub_zeta_r[3:-2],
-        rtol=1e-3,
-        atol=1e-3 * np.nanmean(np.abs(data["B_zeta_r"])),
+        rtol=rtol,
+        atol=atol * np.nanmean(np.abs(data["B_zeta_r"])),
+    )
+    np.testing.assert_allclose(
+        data["B_zeta_rr"][3:-2],
+        B_sub_zeta_rr[3:-2],
+        rtol=rtol,
+        atol=atol * np.nanmean(np.abs(data["B_zeta_rr"])),
+    )
+    np.testing.assert_allclose(
```

Comment:
these B, B_r and B_rr don't make sense to me, isn't `B` and `B_r`, `B_rr` the vector magnetic field? so they should not be compared against the magnetic field magnitude's radial derivative? I am unsure how this even is passing the test....